api-channel-issue-tracker/src/main/java/com/synopsys/integration/alert/api/channel/issue/IssueTrackerChannelLock.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * api-channel-issue-tracker
+ *
+ * Copyright (c) 2022 Synopsys, Inc.
+ *
+ * Use subject to the terms and conditions of the Synopsys End User Software License and Maintenance Agreement. All rights reserved worldwide.
+ */
+package com.synopsys.integration.alert.api.channel.issue;
+
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class IssueTrackerChannelLock {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    // TODO: make this configurable per channel.
+    // 30 minutes
+    public static final int DEFAULT_TIMEOUT_SECONDS = 1800;
+    private static final int PERMITS = 1;
+    private String lockName;
+    private Semaphore lock;
+
+    public IssueTrackerChannelLock(String lockName) {
+        this.lockName = lockName;
+        this.lock = new Semaphore(PERMITS, false);
+    }
+
+    public boolean getLock(int timeoutSeconds) {
```

Comment:
wraps around a semaphore with one permit to allow only 1 person to acquire the lock.  Each issue tracking channel will have an instance of the lock to prevent the duplicate creation of jira tickets.

The introduction of this class also initially indicates that the team should spend more time on the processing of issue tracking messages to determine if we can refine or improve the processing to omit the need for this.